packages/react/src/components/button.tsx

```diff
@@ -106,128 +104,104 @@ export function UploadButton<
 
   const fileInputRef = useRef<HTMLInputElement>(null);
   const labelRef = useRef<HTMLLabelElement>(null);
-  const [uploadProgressState, setUploadProgress] = useState(
+  const [uploadProgress, setUploadProgress] = useState(
     $props.__internal_upload_progress ?? 0,
   );
-  const [files, setFiles] = useState<File[]>([]);
-  const [isManualTriggerDisplayed, setIsManualTriggerDisplayed] =
-    useState(false);
```

Comment:
this state didn't make any sense to me. it's the same as `mode === "manual" && files.length > 0`. Could declare as a variable but just inlined it in the place it was used